app/models/transient_registration.rb

```diff
@@ -0,0 +1,37 @@
+class TransientRegistration
+  include Mongoid::Document
+  include CanHaveRegistrationAttributes
+
+  # TODO: Add state machine
+
+  def renewal_attributes
+    registration = Registration.where(reg_identifier: reg_identifier).first
+    # Don't return object IDs as Mongo should generate new unique ones
+    registration.attributes.except("_id")
+  end
+
+  validate :valid_reg_identifier?
+  validate :no_renewal_in_progress?
+  validate :registration_exists?
+
+  private
+
+  def valid_reg_identifier?
+    # Make sure the format of the reg_identifier is valid to prevent injection
+    # Format should be CBDU, followed by at least one digit
+    return unless reg_identifier.blank? || !reg_identifier.match?(/^CBDU[0-9]+$/)
```

Comment:
Unfortunately we can't rely on all upper tier registrations starting with **CBDU**. Run the following query in MongoDb and you'll see we have Upper tier paid for registrations that start **CBDL**.

```javascript
db.registrations.find({
  $and: [
    { 'tier': { $exists: true } },
    { 'tier': 'UPPER'},
    { 'regIdentifier': /^CBDL/ }
  ]
})
```
So this will need to be updated to account for this.